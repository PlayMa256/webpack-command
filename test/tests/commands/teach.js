const { readFileSync: read } = require('fs');

const { test } = require('../../util');

const TeachCommand = require('../../../lib/commands/TeachCommand');

const dataFilePath = resolve('./data/commands.json');
const testCommandPath = resolve('./test/fixtures/commands/TestCommand.js');

const { TeachCommandError } = TeachCommand;
const command = new TeachCommand();

test('teach command', module, () => {
  it('should throw for no args', () => {
    const stub = () => command.run({ flags: {} });

    expect(stub).toThrowErrorMatchingSnapshot(TeachCommandError);
  });

  it('should throw for no --command value', () => {
    const flags = { command: true };
    const stub = () => command.run({ flags });

    expect(stub).toThrowErrorMatchingSnapshot(TeachCommandError);
  });

  it('should throw for no --module', () => {
    const flags = { command: 'test' };
    const stub = () => command.run({ flags });

    expect(stub).toThrowErrorMatchingSnapshot(TeachCommandError);
  });

  it('should throw for no --module value', () => {
    const flags = { command: 'test', module: true };
    const stub = () => command.run({ flags });

    expect(stub).toThrowErrorMatchingSnapshot(TeachCommandError);
  });

  it('should throw for missing --module', () => {
    const flags = { command: 'test', module: 'missing-module' };
    const stub = () => command.run({ flags });

    expect(stub).toThrowErrorMatchingSnapshot(TeachCommandError);
  });

  it('should learn module', () => {
    const flags = { command: 'test', module: testCommandPath };
    command.run({ flags });

    expect(read(dataFilePath, 'utf-8')).toMatchSnapshot();
  });

  it('should throw for duplicate modules', () => {
    const flags = { command: 'test', module: testCommandPath };
    const stub = () => command.run({ flags });

    expect(stub).toThrowErrorMatchingSnapshot(TeachCommandError);
  });

  it('should forget module', () => {
    const flags = { command: 'test', forget: true };
    command.run({ flags });

    expect(read(dataFilePath, 'utf-8')).toMatchSnapshot();
  });
});
