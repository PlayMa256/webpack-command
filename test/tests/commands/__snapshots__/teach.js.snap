// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teach command > should forget module #0 1`] = `
"{
  \\"help\\": \\"./HelpCommand\\",
  \\"teach\\": \\"./TeachCommand\\"
}"
`;

exports[`teach command > should learn module #0 1`] = `
"{
  \\"help\\": \\"./HelpCommand\\",
  \\"teach\\": \\"./TeachCommand\\",
  \\"test\\": \\"<PROJECT_ROOT>/test/fixtures/commands/TestCommand.js\\"
}"
`;

exports[`teach command > should throw for duplicate modules #0 1`] = `"The 'test' command was already added via <PROJECT_ROOT>/test/fixtures/commands/TestCommand.js. Use \`webpack teach --command test --forget\` and try again."`;

exports[`teach command > should throw for missing --module #0 1`] = `"The command module 'missing-module' was not found."`;

exports[`teach command > should throw for no --command value #0 1`] = `"The value of --command must be a String."`;

exports[`teach command > should throw for no --module #0 1`] = `"The --module flag is required."`;

exports[`teach command > should throw for no --module value #0 1`] = `"The value of --module must be a String."`;

exports[`teach command > should throw for no args #0 1`] = `"The --command flag is required."`;
